src/types.rs

```diff
@@ -1388,3 +1388,61 @@ define_bounded_integers! {
     BoundedI64 i64 signed -> i64,
     BoundedIsize isize signed -> isize,
 }
+
+//Indexing operations on [T; N], Vec<T> and VecDeque<T> for BoundedUsize
+#[cfg(all(feature = "std", feature = "types"))]
+impl<const MIN: usize, const MAX: usize, const N: usize, T> std::ops::Index<BoundedUsize<MIN, MAX>>
+    for [T; N]
```

Comment:
Could this be generalized to implementing `Index` for `[T]`? If that implementation is present, we get the array implementation for free because of [this implementation in the core library](https://doc.rust-lang.org/stable/std/ops/trait.Index.html#impl-Index%3CI%3E-for-%5BT%3B%20N%5D).